### PR TITLE
[5.6] Use Carbon 1.26

### PR DESCRIPTION
**Pull request by kylekatarnls**
Now that Carbon\Carbon supports macros and JSON, Illuminate\Support\Carbon should be emptied, then it would be the end of compatibility issues.

### CODE REVIEW

**Comment by 36864** [comment]
At a minimum this needs to require "^1.26.2" due to the problems with 1.26/1.26.1.

This is also probably a breaking change for anyone extending Support\Carbon into their own class and should target 5.7

**Comment by kylekatarnls** [comment]
Nope, because by empty the Laravel Class (no longer needed), there is no longer conflict.

And this should also be done in master for 5.7

File: src/Illuminate/Support/Carbon.php
Comment (tillkruss) [inline src/Illuminate/Support/Carbon.php]: This is a breaking change. Either send it to v5.7, or don't remove `Macroable`.

**Review by tillkruss — CHANGES_REQUESTED** [review]

File: src/Illuminate/Support/Carbon.php
Comment (kylekatarnls) [reply src/Illuminate/Support/Carbon.php]: No, it's not since compatible methods are now in Carbon itself.

Comment (tillkruss) [reply src/Illuminate/Support/Carbon.php]: You're saying that peoples existing macros won't break with this change?

**Review by tillkruss — CHANGES_REQUESTED** [review]

**Comment by devcircus** [comment]
It should not break existing apps. Lucas tried to push this functionality to Carbon before sending to Laravel. It was rejected. Now Carbon maintainers have changed their minds and copied the functionality over to the their package.

The macro and mixin methods will continue to work, however I prefer keeping it in Illuminate\Support.

**Comment by 36864** [comment]
This is still a breaking change. Any code relying explicitly on the presence of the Macroable trait in the class will break.
```
>>> class_uses(Carbon\Carbon::class);
=> []
>>> class_uses(Illuminate\Support\Carbon::class);
=> [
     "Illuminate\Support\Traits\Macroable" => "Illuminate\Support\Traits\Macroable",
   ]
```

**Comment by kylekatarnls** [comment]
@36864 Seems very specific, but indeed. This argument seems sufficient to me for waiting for a minor release (such as 5.7).

I know it's easy to say now that it's done, but such ability should be tested with interfaces rather than traits. Maybe on the long term, a MacroableInterface should be added (maybe submitted to https://www.php-fig.org/psr/) with macro, mixin, hasMacro. And this interface should be used to test if macro stuff is available rather than testing the trait has been used. Don't you think so?

**Comment by taylorotwell** [comment]
No plans to make this change on 5.6.

**Comment by kylekatarnls** [comment]
Legit, but does #23896 can be reconsidered so?

**Comment by X-Coder264** [comment]
@kylekatarnls I think you should resubmit #23896 because Taylor usually doesn't come back to check for new comments in the PRs which are closed (such as this one).

**Comment by SolStis86** [comment]
https://soundcloud.com/hardcore-messiah/dougal-b2b-gammer-slammin-vinyl-nye-2007 1:13:00 -> exciting. Beyond this... WHY is it happening. There is no explanation as to the failing tests.

**Pull request closed without merging**
